h5grove/fastapi_utils.py

```diff
@@ -51,6 +51,8 @@ async def add_base_path(file):
     filepath = f"{settings.base_dir}/{file}" if settings.base_dir else file
     if not os.path.isfile(filepath):
         raise HTTPException(status_code=404, detail="File not found!")
+    if not os.access(filepath, mode=os.R_OK):
+        raise HTTPException(status_code=403, detail="Cannot read file!")
```

Comment:
I would just go for a more specific error message, as "cannot read file" is a bit generic.

Reply:
Interesting proposals, thanks ! I will stick with `os.access` to avoid the extra file open/close :sweat_smile:

Reply:
Note that having a common `H5GroveError` may be interesting for #72 so I may end up adapting this in the end

Reply:
Yes, that was also the idea behind it, because the error handling is duplicated each time you make requests handlers: in each `*_utils.py`, in jupyterlab-h5web and I also have a copy it in a project..

Reply:
Let's continue the discussion on the incoming PR for #72 then :+1: